src/s390_ecc.c

```diff
@@ -1743,16 +1743,22 @@ struct {				\
 #ifdef ICA_FIPS
 			if (fips & ICA_FIPS_MODE) {
 				fc |= 0x80; /* deterministic signature */
-				RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+				do {
+					/* A cc=2 occurs if the random number is not invertible.
+					 * Loop in this case. */
+					RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
```

Comment:
I guess we should check the error of RAND_bytes() here. For the FIPS case it is really essential that the random number generator did work, so we should error out if not.

For the other cases, RAND_bytes is just used to get a better random quality for the nonce and it does not really matter if it fails. KDSA will anyway generate a random nonce internally.

Reply:
ok, done.